behave/features/steps/public_exit_page.py

```diff
@@ -0,0 +1,37 @@
+from behave import step
+
+from selenium.webdriver.common.by import By
+from selenium.webdriver.common.action_chains import ActionChains
+from selenium.webdriver.common.keys import Keys
+
+from helper.constants import BBC_WEBSITE
+
+
+@step('The "Exit this page" button is on the page and I click it')
+def step_impl_click_exit_page(context):
+    href_bbc_address = context.helperfunc.find_by_xpath(
+        "//main/div/div/a[contains(text(), 'Exit this page')]"
+    ).get_attribute("href")
+
+    assert href_bbc_address == BBC_WEBSITE
+
+    context.helperfunc.click_button(By.CLASS_NAME, "govuk-exit-this-page")
+
+
+@step('I press the "{keypress}" key {amount_of_time} times on the keyboard')
+def step_impl_keypress_multiple_times(context, keypress, amount_of_time):
+    key_to_press = (
+        Keys.SHIFT if keypress == "shift" else Keys.TAB if keypress == "tab" else None
+    )
+
+    for _ in range(int(amount_of_time)):
+        if key_to_press:
+            ActionChains(context.helperfunc.driver()).send_keys(key_to_press).perform()
+
+    if keypress == "tab":
+        ActionChains(context.helperfunc.driver()).send_keys(Keys.ENTER).perform()
```

Comment:
Just a query about the desired behave best practices.

This will work, however, the implementation doesn't match the feature line.

Should there be a separate step for:
```
@step('I press the "{key:w}" key')
def step_impl_keypress(context, key):
    key_map = {'enter': Keys.ENTER}
    
    if key in key_map:
        key_to_press = key_map[key]
        ActionChains(context.helperfunc.driver()).send_keys(key_to_press).perform()
```

Either implementation works, however, this implementation should be better for future proofing the code and creating re-usable test patterns.